### ISSUE TITLE
Trace ID not Maintained over Multiple Service Hops with Express/Fetch

### ISSUE THREAD

**Issue by astorm** [openzipkin/zipkin-js#501]
## Describe the Bug

I have a system with a "three hop" service call.  The service calls are made with `node-fetch`, the services implemented with `express`.

Instead of seeing everything as a single trace, Zipkin sees two traces.

## Steps to Reproduce

0. Start zipkin `docker run -d -p 9411:9411 openzipkin/zipkin`
1. Clone the repo https://github.com/astorm/zipkin-example
2. Run `npm install`
3. Run `npm run start`
4. Confirm services start
   ```
        $ npm run start

        > zipkin@1.0.0 start /private/tmp/zipkin/zipkin-example
        > node src/index.js

        Example app listening at http://localhost:3000
        Example service listening at http://localhost:3001
        Example service listening at http://localhost:3002
   ```
5. Run `curl http://localhost:3000/main`
6. View most recent trace in ZipKin

## Expected Behavior

Expected Behavior: I expect to see a single trace with three spans: from `service-main` to `service-hello` to `service-goodbye`

Actual Behavior: I see a two traces with two spans each: 1. From `service-main` to `service-hello` 2. From `service-hello` to `service-goodbye`

See README in https://github.com/astorm/zipkin-example for more information on a reproduction.

> The best is to spend some time to write a failing test. Bugs with tests get fixed and stay fixed.

I would like to write a failing test.  However, I'm new to Zipkin's and it is not clear what package(s) and/or code is responsible for maintaining and setting a trace id, so I'm unsure how to write a test (other than the repo above) for this. Guidance here would be rad.

## Additional Information

A helpful human suggested I might be able to fix this by changing

    "zipkin-context-cls": "0.21.0",

to

    "zipkin-context-cls": "0.19.2-alpha.7",

due [to this issue/PR](https://github.com/openzipkin/zipkin-js/pull/498).  However, this did not help my specific problem.

**Comment by astorm** [comment]
Ughhhhhhh,  apologies for the distraction, I've had some brain rot about how express works.  Closing out -- this isn't an issue with Zipkin. 

For anyone who comes across this later and might have done the same thing, [the problem is here](https://github.com/astorm/zipkin-example/blob/7752e7b9e7d7ff24b02c76669608b466dda00a39/src/service-main.js#L21) 

I had added the express middleware _after_ my call to `app.get`
    
    app.get('/main', async function (req, res) {
        //...
    })

    app.use(zipkinMiddleware({tracer}));

When ordering, Express ([apparently](https://derickbailey.com/2016/05/09/in-what-order-does-my-express-js-middleware-execute/)) makes no distinction between `app.use` and `app.[HTTPMETHOD]` middlwares.  So in my example application, the zipkin middlware never gets a chance to run.
